Sprint 09

Ka1der opened the pull request.

Ka1der committed:
Test commit for PR_2

Ka1der committed:
Сверстал окно ЛК и добавил outlet'ы и action'ы

Ka1der committed:
Сверстал окно ЛК и добавил outlet'ы и action'ы

Ka1der committed:
Сделал Tab Bar Controller

Ka1der committed:
Сделал SingleImageViewController

Ka1der committed:
Сделал SingleImageViewController

Ka1der committed:
Сделал возможность открытия изображений на новом экране

Ka1der committed:
Добавил кнопку backWard

Ka1der committed:
Добавил кнопку backWard

Ka1der committed:
Добавил scrollView

Ka1der committed:
Реализовал функциональность зума в приложении.

Ka1der committed:
Написал экран ЛК кодом и добавил алерт выхода

Ka1der committed:
Исправил первый экран который будет открывать TabBarController

Ka1der committed:
Прописал MARK'и

DanilaG commented on Image Feed/ImagesList/ProfileViewController.swift:
```diff
@@ -0,0 +1,141 @@
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = "@ekaterina_nov"
+        label.font = UIFont.systemFont(ofSize: 16, weight: .regular)
+        label.textColor = UIColor(named: "#AEAFB4")
```
Давай добавим цвет в Assets

DanilaG commented on Image Feed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,82 @@
+    // MARK: - Properties
+    
+    var image: UIImage? {
+        
```
Обычно, тут редко когда делают отступ)

DanilaG commented on Image Feed/ImagesList/SingleImageViewController.swift:
```diff
@@ -0,0 +1,82 @@
+            view.layoutIfNeeded()
+            let visibleRectSize = scrollView.bounds.size
+            let imageSize = image.size
+            let hScale = visibleRectSize.width / imageSize.width
+            let vScale = visibleRectSize.height / imageSize.height
```
Мало вероятно, но `imageSize.width` или `imageSize.height` может оказаться == 0, тогда получаем падение приложения при попытки делить на 0. Давай добавим проверку деления на 0

Ka1der committed:
Исправил шрифт лейбла с "Hello, world!" на 13 пикселей
Исправил шрифт лейбла с "@ekaterina_nov" на 13 пикселей
Исправил размеры кнопки выхода на 24х24 пиксеоя
Исправл цвет кнопки выхода
Исправил размеры кнопки поделяться на 51х51 пиксель
Исправл фон  launch screen

Ka1der committed:
Прописал наиманования для цветов - YPGray и YPRed

Ka1der committed:
Прописал наиманования для цветов - YPGray и YPRed

Ka1der committed:
Добавил проверку imageSize.width или imageSize.height на 0

Ka1der commented:
Привет ревьюер!
Спасибо за замечания. 
Отступ между MARK и кодом я не исправлял потому мне пока что так проще воспринимать код. 
Все остальное исправлено)

Ka1der committed:
Загрузил заново иконку выхода. Теперь отображается нормально
